backend/inspirehep/accounts/views.py

```diff
@@ -57,8 +55,15 @@ def sign_up_user():
         return jsonify(data_current_user), 200
 
 
-@blueprint.route("/login", methods=["POST"])
+@blueprint.route("/login", methods=["GET"])
 def login():
+    session["next_url"] = request.args.get("next")
+    session["signup_url"] = request.args.get("signup_url")
+    return redirect("/api/oauth/login/orcid")
```

Comment:
why do we need this new `view` ?

Reply:
Because we need to save `next_url` and `signup_url` so that we can do the redirection later. If we just pass them as query params to `oauthclient` they're ignored. `oauthclient` only reads `next_url` from config.

Reply:
so if we didn't need `signup_url`, we wouldn't need this?

Reply:
We would still need it for the `next_url` since it's dynamic

Reply:
I haven't checked deeply and `invenio-oauthclient`'s code is not known for its simplicity so I might be wrong but it shouldn't be like that.

Reply:
https://github.com/inveniosoftware/invenio-oauthclient/blob/e7e4804260d4a4058414415fcb90d024a4dfa8ea/tests/test_views.py#L48

Reply:
If I do `/api/oauth/login/orcid?next=/jobs` and don't configure any `authorized_redirect_url` what I get after logging in is `/?message=Successfully+authorized.&code=200#show_login`.

Reply:
The issue here is that they don't handle the ``next`` url https://github.com/inveniosoftware/invenio-oauthclient/blob/4e1acca086cf894a1a0184e96a30c401754f7317/invenio_oauthclient/handlers/rest.py#L40 in the default handler, we can override it https://github.com/inveniosoftware/invenio-oauthclient/blob/4e1acca086cf894a1a0184e96a30c401754f7317/invenio_oauthclient/config.py#L105

Reply:
Like it's in the refinement, no?

Ok so seems like there are 2 options:

either we create those dummy views to redirect around which seems extra but kinda easy to read what's going on.

or we just override the handler to use `next_url`.

Which do you think it's better? @drjova @miguelgrc

Reply:
let's have the control, they might change handlers again

Reply:
alright then, @miguelgrc you can go like you did initially.

Reply:
It should be ready like this then :)